scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -496,6 +502,18 @@ object Defn {
     @replacedField("4.6.0") final def paramss: List[List[Term.Param]] =
       paramClauses.map(_.values).toList
   }
+  @ast class InterleavedDef(
```

Comment:
unfortunately, not every tree has an origin, and hence token positions.

another option is to add an explicit offset field to param clauses, but that might be a bit of a hack.

also, tparam is not a sequence but a single one, although we could just return the first one.

thoughts?

Reply:
> unfortunately, not every tree has an origin, and hence token positions.

That would be for transformed trees, no? Having `InterleavedDef` seems too much of a change and we would need to handle it everywhere we would handle Def, which might be a lot of additional cases.

Reply:
this applies to constructed (as opposed to parsed) trees. wouldn't be surprised that scalafix or any other ast-based application will not be able to use that.

Reply:
What we basically want here is having two fields replaced by one, I think that can't be handled by the current `repalacedField` no?

The other possibility is to have a way to match `InterleavedDef` with the standard Def unapply maybe? And could we make InterleavedDef extend Def? 

Or, we could have Def as a trait extended by OldDef and InterleavedDef?

Just throwing the ideas out there. Essentially an ideal situation would be that if someone enables clause itnerleaving then the same code that work without that dialect still works, otherwise it will all silently fail unless there are proper tests in place.

Reply:
@tgodzik figured out how to do it. a few intermediate PRs for you to review, before i update this one.